addons/l10n_ar/models/account_move.py

```diff
@@ -360,11 +361,62 @@ def _l10n_ar_get_invoice_totals_for_report(self):
             for subtotal in tax_totals['subtotals']
             for tax_group in subtotal['tax_groups']
         }
-        tax_group_ids_to_exclude = self.env['account.tax.group'].browse(tax_group_ids).filtered('l10n_ar_vat_afip_code').ids
+        tax_group_ids_to_exclude = self.env['account.tax.group']\
+            .browse(tax_group_ids)\
+            .filtered(lambda tax_group: (
+                not self._l10n_ar_is_tax_group_other_national_ind_tax(tax_group)
+                and not self._l10n_ar_is_tax_group_vat(tax_group)
+            )).ids
         if tax_group_ids_to_exclude:
-            return self.env['account.tax']._exclude_tax_groups_from_tax_totals_summary(tax_totals, tax_group_ids_to_exclude)
+            tax_totals = self.env['account.tax']._exclude_tax_groups_from_tax_totals_summary(tax_totals, tax_group_ids_to_exclude)
         return tax_totals
 
+    def _l10n_ar_get_invoice_custom_tax_summary_for_report(self):
```

Comment:
@zaoral I separate the computation of this from the tax_totals. The taxes aggregators avoid unconsistencies between grouping keys and are doing all aggregations for you.

Reply:
perfect